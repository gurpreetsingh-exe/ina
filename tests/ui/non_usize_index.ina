// COMPILE_FAIL

fn main() {
    let xs = [1, 2, 3];
    // error
    xs[20.];
    // error
    xs[xs];
    let i: i32 = 0;
    // error
    xs[i];
    // ok
    xs[0];
    let i: usize = 0;
    xs[i];
}
