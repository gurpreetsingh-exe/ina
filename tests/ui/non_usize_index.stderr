error: mismatch types
   ╭─[tests/ui/non_usize_index.ina:6:8]
   │
 6 │     xs[20.];
   │        ─┬─
   │         ╰── expected `usize`, found `float`
   ∙
───╯
error: mismatch types
   ╭─[tests/ui/non_usize_index.ina:8:8]
   │
 8 │     xs[xs];
   │        ╶╮
   │         ╰── expected `usize`, found `[integer]`
   ∙
───╯
error: mismatch types
    ╭─[tests/ui/non_usize_index.ina:11:8]
    │
 11 │     xs[i];
    │        ┬
    │        ╰─ expected `usize`, found `i32`
    ∙
────╯
